rosidl_generator_cpp/resource/msg__struct.hpp.em

```diff
@@ -81,55 +86,112 @@ struct @(spec.base_type.type)_
+# http://design.ros2.org/articles/generated_interfaces_cpp.html#constructors
+# for a detailed explanation of the different _init parameters.
+init_list, alloc_list, member_list = create_init_alloc_and_member_lists(spec)
+
+def generate_default_string(membset):
+    strlist = []
+    for member in membset.members:
+        if member.default_value is not None:
+            if member.num_prealloc > 0:
+                strlist.append('this->%s.resize(%d);' % (member.name, member.num_prealloc))
+            if isinstance(member.default_value, list):
+                if member.default_value:
+                    if all(v == member.default_value[0] for v in member.default_value):
+                        strlist.append('std::fill(this->%s.begin(), this->%s.end(), %s);' % (member.name, member.name, member.default_value[0]))
+                    else:
+                        for index, val in enumerate(member.default_value):
+                            strlist.append('this->%s[%d] = %s;' % (member.name, index, val))
+            else:
+                strlist.append('this->%s = %s;' % (member.name, member.default_value))
+
+    return strlist
+
+def generate_zero_string(membset, fill_args):
+    strlist = []
+    for member in membset.members:
+        if isinstance(member.zero_value, list):
+            if member.num_prealloc > 0:
+                strlist.append('this->%s.resize(%d);' % (member.name, member.num_prealloc))
+            if member.zero_need_array_override:
+                strlist.append('this->%s.fill(%s {%s});' % (member.name, member.type, fill_args))
+            else:
+                strlist.append('std::fill(this->%s.begin(), this->%s.end(), %s);' % (member.name, member.name, member.zero_value[0]))
+        else:
+            strlist.append('this->%s = %s;' % (member.name, member.zero_value))
+    return strlist
 }@
-@[  if alloc_name == '_alloc']@
-// *INDENT-ON*
-@[  end if]@
+  explicit @(spec.base_type.type)_(rosidl_generator_cpp::MessageInitialization _init = rosidl_generator_cpp::MessageInitialization::ALL)
+@[if init_list]@
+  : @(',\n    '.join(init_list))
+@[end if]@
   {
-@# generate constructor body
-@[  if alloc_name and not used_alloc_name]@
-    (void)@(alloc_name);
-@[  end if]@
-@{
-for field in spec.fields:
-    # default values of dynamic arrays with allocators if available
-    if alloc_name and field.type.is_array and \
-            (field.type.array_size is None or field.type.is_upper_bound) and \
-            field.type.is_primitive_type() and field.default_value is not None:
-        cpp_value = value_to_cpp(field.type, field.default_value)
-        print('    %s = %s;' % (field.name, cpp_value))
-}@
+@[if not member_list]@
+    (void)_init;
+@[end if]@
+@[for membset in member_list]@
+@[ if membset.members[0].default_value is not None]@
+    if (rosidl_generator_cpp::MessageInitialization::ALL == _init ||
+        rosidl_generator_cpp::MessageInitialization::DEFAULTS_ONLY == _init) {
```

Comment:
This line doesn't pass `uncrustify`. It needs to look like lines 151-152.

Reply:
Good point.  It should be fixed b41a4d00

Reply:
The generated code still doesn't pass uncrustify. The open curly brace needs to be on the next line.

The `rosidl_generate_interfaces()` in the testing section should pass `ADD_LINTER_TESTS` to automatically lint the generated code.

Reply:
I'm not sure why it is passing locally, but not when run elsewhere.  I'll add the `ADD_LINTER_TESTS` to the code and see what turns up.

Reply:
@clalancette It seems that you addressed only the comment on this line and not the "same below" part, it still fails uncrustify locally